components/camel-zeebe/src/test/java/org/apache/camel/component/zeebe/ZeebeConsumerIntegrationTest.java

```diff
@@ -0,0 +1,184 @@
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.zeebe;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.component.zeebe.model.DeploymentRequest;
+import org.apache.camel.component.zeebe.model.DeploymentResponse;
+import org.apache.camel.component.zeebe.model.JobRequest;
+import org.apache.camel.component.zeebe.model.JobWorkerMessage;
+import org.apache.camel.component.zeebe.model.ProcessDeploymentResponse;
+import org.apache.camel.component.zeebe.model.ProcessRequest;
+import org.apache.camel.component.zeebe.model.ProcessResponse;
+import org.apache.camel.test.junit5.CamelTestSupport;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Tag("standalone")
+@MockitoSettings(strictness = Strictness.LENIENT)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ZeebeConsumerIntegrationTest extends CamelTestSupport {
+
+    public static final String TEST_1_DEFINITION_BPMN = "test1_definition.bpmn";
+    private ZeebeComponent component;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeAll
+    void initAll() throws Exception {
+        createComponent();
+        component.doStart();
+
+        DeploymentRequest deployProcessMessage = new DeploymentRequest();
+        deployProcessMessage.setName(TEST_1_DEFINITION_BPMN);
+        deployProcessMessage
+                .setContent(this.getClass().getClassLoader().getResourceAsStream("data/test1_definition.bpmn").readAllBytes());
+
+        DeploymentResponse deploymentResponse = component.getZeebeService().deployResource(deployProcessMessage);
+
+        ProcessRequest processRequest = new ProcessRequest();
+        processRequest.setProcessId(((ProcessDeploymentResponse) deploymentResponse).getBpmnProcessId());
+        ProcessResponse processResponse = component.getZeebeService().startProcess(processRequest);
+    }
+
+    @BeforeEach
+    void init() throws Exception {
+
+    }
+
+    @Test
+    public void shouldProcessJobWorkerMessage() throws Exception {
+        MockEndpoint workerMock = getMockEndpoint("mock:jobWorker");
+        workerMock.expectedMinimumMessageCount(1);
+
+        for (int i = 0; i < 10; i++) {
+            if (!workerMock.getExchanges().isEmpty()) {
+                break;
+            }
+            TimeUnit.SECONDS.sleep(2);
```

Comment:
Please, avoid `sleeps` on the test code. Our tests already consume way too much time and we have been trying to reduce this. If some kind of synchronization is needed, I'd recommend looking at using Awaitility to do so.